homeassistant/components/media_player/sonos.py

```diff
@@ -150,8 +150,12 @@ def setup_platform(hass, config, add_devices, discovery_info=None):
             _LOGGER.warning("No Sonos speakers found")
             return
 
-        hass.data[DATA_SONOS] = [SonosDevice(p) for p in players]
-        add_devices(hass.data[DATA_SONOS], True)
+        # Add coordinators first so they can be queried by slaves
+        coordinators = [SonosDevice(p) for p in players if p.is_coordinator]
+        slaves = [SonosDevice(p) for p in players if not p.is_coordinator]
```

Comment:
I feel there should be a simple way to partition a list based on a predicate without traversing the list twice, but I don't know that way.

Reply:
`itertools.groupby` perhaps?